contracts/protocol/facets/ConfigHandlerFacet.sol

```diff
@@ -558,6 +611,33 @@ contract ConfigHandlerFacet is IBosonConfigHandler, ProtocolBase {
         return address(DiamondLib.diamondStorage().accessController);
     }
 
+    /**
+     * @notice Sets the price ranges for a specific token.
+     *
+     * @param _tokenAddress - the address of the token
+     * @param _priceRanges - array of price ranges for the token
+     */
+    function setTokenPriceRanges(address _tokenAddress, uint256[] calldata _priceRanges) internal {
+        // Set new price ranges
+        for (uint256 i = 0; i < _priceRanges.length; i++) {
+            protocolFees().tokenPriceRanges[_tokenAddress].push(_priceRanges[i]);
+        }
+    }
+
+    /**
+     * @notice Sets the fee percentages for a specific token and price ranges.
+     *
+     * @param _tokenAddress - the address of the token
+     * @param _feePercentages - array of fee percentages corresponding to each price range
+     */
+    function setTokenFeePercentages(address _tokenAddress, uint256[] calldata _feePercentages) internal {
+        // Set the fee percentages for the token
+        for (uint256 i = 0; i < _feePercentages.length; i++) {
+            checkMaxPercententage(_feePercentages[i]);
+            protocolFees().tokenFeePercentages[_tokenAddress].push(_feePercentages[i]);
```

Comment:
Similar to above, the array can be stored with `protocolFees().tokenFeePercentages[_tokenAddress] = _feePercentages;`

We just need to keep the loop because of `checkMaxPercententage`